core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -88,11 +88,9 @@ public TableMetadata refresh() {
       if (currentMetadataWasAvailable) {
         LOG.warn("Could not find the table during refresh, setting current metadata to null", e);
       }
-
       currentMetadata = null;
       currentMetadataLocation = null;
       version = -1;
-      shouldRefresh = false;
```

Comment:
I think this needs to set `shouldRefresh` to true. If `refresh` was called directly, then `shouldRefresh` may not already be true. But we want it to be true so that this will not return metadata until a successful refresh.

Reply:
Also, since `current` is called in `commit`, I think we will need to only set this in the case when `currentMetadataWasAvailable`. That way, creating a table will still work the same way.